File: Level 2/타겟 넘버/solution.js
// DFS, 깊이 우선 탐색
function solution(numbers, target) {
  let answer = 0;

  const dfs = (sum, index) => {
    if (numbers.length === index) {
      if (target === sum) {
        answer++;
      }
      return;
    }

    dfs(sum + numbers[index], index + 1);
    dfs(sum - numbers[index], index + 1);
  };

  dfs(0, 0);

  return answer;
}

File: Level 2/타겟 넘버/solution.py
def solution(numbers, target):
    def dfs(total, index):
        count = 0

        if index == len(numbers):
            return 1 if total == target else 0

        count += dfs(total + numbers[index], index + 1)
        count += dfs(total - numbers[index], index + 1)

        return count

    answer = dfs(0, 0)

    return answer
